scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -2091,10 +2102,18 @@ private void sendMessage(final long nanos, final RawData message, final Connecti
 		LOGGER.debug("Sending application layer message to [{}]", message.getEndpointContext());
 
 		Handshaker handshaker = connection.getOngoingHandshake();
-		if (handshaker != null && handshaker.isExpired()) {
-			// handshake expired during Android / OS "deep sleep"
-			// on sending, abort, keep connection for new handshake
-			handshaker.handshakeAborted(new Exception("handshake already expired!"));
+		if (handshaker != null) {
+			if (handshaker.isExpired()) {
+				// handshake expired during Android / OS "deep sleep"
+				// on sending, abort, keep connection for new handshake
+				handshaker.handshakeAborted(new Exception("handshake already expired!"));
+			} else if (handshaker.isProbing()) {
+				if (checkOutboundEndpointContext(message, null)) {
```

Comment:
Not linked to this PR, but maybe a method with a clearer name and without `null` param could make this clearer.